### PR TITLE
Fix some typescript issues

### PR DESCRIPTION
**Pull request by silverwind**
Fixes around 30 or so typescript errors. No runtime changes.

### CODE REVIEW

File: web_src/js/modules/fomantic/dimmer.ts [hunk 1]
```diff
@@ -3,7 +3,7 @@ import {queryElemChildren} from '../../utils/dom.ts';
 
 export function initFomanticDimmer() {
   // stand-in for removed dimmer module
-  $.fn.dimmer = function (arg0, arg1) {
+  $.fn.dimmer = function (arg0: string, arg1: () => void) {
```
Comment (wxiaoguang) [inline web_src/js/modules/fomantic/dimmer.ts]: arg1 could be anything

File: web_src/js/modules/fomantic/dropdown.ts [hunk 2]
```diff
@@ -74,7 +75,7 @@ function delegateOne($dropdown) {
   // the "template" functions are used for dynamic creation (eg: AJAX)
   const dropdownTemplates = {...dropdownCall('setting', 'templates'), t: performance.now()};
   const dropdownTemplatesMenuOld = dropdownTemplates.menu;
-  dropdownTemplates.menu = function(response, fields, preserveHTML, className) {
+  dropdownTemplates.menu = function(response: any, fields: any, preserveHTML: any, className: string) {
```
Comment (wxiaoguang) [inline web_src/js/modules/fomantic/dropdown.ts]: IIRC className is an object

File: web_src/js/modules/fomantic/dimmer.ts (hunk 1 above)
Comment (silverwind) [reply web_src/js/modules/fomantic/dimmer.ts]: ah, right

Comment (silverwind) [inline web_src/js/modules/fomantic/dimmer.ts]:
```suggestion
  $.fn.dimmer = function (arg0: any, arg1: () => void) {
```

File: web_src/js/modules/fomantic/transition.ts [hunk 3]
```diff
@@ -8,13 +8,13 @@ export function initFomanticTransition() {
     'set duration', 'save conditions', 'restore conditions',
   ]);
   // stand-in for removed transition module
-  $.fn.transition = function (arg0, arg1, arg2) {
+  $.fn.transition = function (arg0: string | Record<string, any>, arg1: number, arg2: (el: HTMLElement) => void) {
```
Comment (wxiaoguang) [inline web_src/js/modules/fomantic/transition.ts]: args could be anything

File: web_src/js/modules/fomantic/dimmer.ts [hunk 4]
```diff
@@ -3,7 +3,7 @@ import {queryElemChildren} from '../../utils/dom.ts';
 
 export function initFomanticDimmer() {
   // stand-in for removed dimmer module
-  $.fn.dimmer = function (arg0, arg1) {
+  $.fn.dimmer = function (arg0: any, arg1: () => void) {
```
Comment (silverwind) [inline web_src/js/modules/fomantic/dimmer.ts]:
```suggestion
  $.fn.dimmer = function (arg0: string, arg1: any) {
```

File: web_src/js/modules/fomantic/dropdown.ts (hunk 2 above)
Comment (silverwind) [inline web_src/js/modules/fomantic/dropdown.ts]:
```suggestion
  dropdownTemplates.menu = function(response: any, fields: any, preserveHTML: any, className: Record<string, string>) {
```

Comment (silverwind) [reply web_src/js/modules/fomantic/dropdown.ts]: Right, I see I typed it as such in another place.

File: web_src/js/modules/fomantic/transition.ts (hunk 3 above)
Comment (silverwind) [reply web_src/js/modules/fomantic/transition.ts]: Well, I can type as `any`, it's not like anyone is going to touch these functions.

**Comment by silverwind** [comment]
FYI, in case anyone is wondering about the `no-undef` TODO: Typescript fully covers this rule and recommends it to be disabled. Currently the rule raises false-positives for global type annotations like `JQuery`. So my plan is to fix all `tsc` errors and then enable `tcs` and disable this rule. Until then, type annotations like `JQuery` can only be used with a accompanying `// eslint-disable-line`.

**Review by wxiaoguang — APPROVED** [review]

**Review by delvh — APPROVED** [review]

**Pull request merged**